### ISSUE TITLE
After update, all of previous notes gone

### ISSUE THREAD

**Issue by rico345100** [BoostIO/Boostnote#901]
Please paste some **screenshots** with the **developer tool** open if you report a bug.

I was using this with added more storages which located at "/Users/rico345100/Desktop/development/boots-repo", but after recent update, they're gone.

I can see the files still there but there is no way to load them. How do I load them?

**Comment by Andrehatlo** [comment]
I have the same issue..

**Comment by kazup01** [comment]
Hi I am Boostnote's maintainer. Sorry for inconvenience.
You guys data has NOT deleted. Please relink the file from preferences.
<img width="700" alt="28759997-4235604a-75de-11e7-9560-8ff78c8a8939" src="https://user-images.githubusercontent.com/8602615/30795918-d857ac9c-a20a-11e7-97dd-1885ac28f373.png">

**Comment by Andrehatlo** [comment]
.cson
<img width="586" alt="screen shot 2017-09-25 at 09 04 03" src="https://user-images.githubusercontent.com/23061393/30796000-843d78b0-a1d0-11e7-8bee-9a213fb1fe42.png">

If im not mistaken, the application saves all files as `.cson`.
Searching through `all my files` if clearly tell that all my `.cson files are gone from my system.

**Comment by kazup01** [comment]
Hi @Andrehatlo , thanks for your replying.
Please select the folder that contains .cson files, not each .cson files.
<img width="715" alt="screen shot 0029-09-25 at 4 17 05 pm" src="https://user-images.githubusercontent.com/8602615/30796493-48c85538-a20d-11e7-9a3d-b10491916638.png">

**Comment by davesnx** [comment]
I got the same issue, and nothing in the folder. Each cson file I supose that is each note. The Boostnote folder appartenly has been created when I update. Any other place where could been saved?

**Comment by davesnx** [comment]
Nothing in `Documents/` neither. 
Any other place where Boostnote was saving it? @kazup01

**Comment by blork-mtl** [comment]
FFS I just lost over an hour of my time to this, on a busy day with a deadline. It took me forever to figure out how to re-link to my storage -- the ONLY way was to unlink and then start from scratch. No option to just edit the storage setting.

Then I had to REBUILD my UI preferences, because my preferences were blown away in this update.

So yes, I'm pissed. Especially because I didn't even have the option to delay the update. 

MY RECOMMENDATIONS:

- Give the user the option to refuse an update ("update later")

- Fix whatever problem causes preferences to get blown away when there's an update. (Seriously, this should be your *top priority*.)

- Let users select or edit their storage by BROWSING instead of having to type in the file path.

:-/

**Comment by kazup01** [comment]
I'm terribly sorry for your inconvenience. @autobrain-stanley

If you didn't set anything, file will be created here. @davesnx
<img width="605" alt="screen shot 0029-09-26 at 12 35 10 pm" src="https://user-images.githubusercontent.com/8602615/30841511-ccf3c9aa-a2b7-11e7-9f68-18cbd8b91712.png">

**Comment by davesnx** [comment]
Yes, the file is created but all my old notes are gone. @kazup01

**Comment by kazup01** [comment]
Notes are not delete automatically. Please search for files already you made. @davesnx

**Comment by aweary** [comment]
@kazup01 you may not have intended to delete the notes, but I also found that all my notes were deleted after the forced update.

**Comment by davesnx** [comment]
Hey @kazup01,

I think I didn't explain the problem well enough. I had 100 notes on Boostnote before the update, and after that were only 2 (the default ones). I can't find the old notes and I suspect that are gone.

Nothing on the folder `Documents/Boostnote` neither `Documents/Boostnote/notes`. Nothing on `All files > Find *.cson`.

Anyway I could recover my data? 

Thanks.

**Comment by kazup01** [comment]
Hi @davesnx , thank you for the detailed information. Again, my apologies for inconvenience.
Have you also search for files other than `/Documents`?

My computer(macOC Sierra v10.12.6) also could not find the .cson files on the search, but I have those files. (I moved these files to Google drive and save them.)
<img width="823" alt="screen shot 0029-09-27 at 11 09 39 am" src="https://user-images.githubusercontent.com/8602615/30892697-ed40c650-a374-11e7-9492-7dcec6166718.png">

**Comment by kazup01** [comment]
Hi @aweary, sorry for inconvenience. What is "delete"? Notes are not deleted automatically even if these are unlinked.

**Comment by kazup01** [comment]
I'm terribly sorry for you guys inconvenience again. I think we must fix this problem.
Could you teach me about you guys dev environment?
In my computer didn't this bug occurred. macOS Sierra v10.12.6

@rico345100  @Andrehatlo @davesnx @autobrain-stanley @aweary

**Comment by davesnx** [comment]
Hi @kazup01, macOS Sierra 10.12.3.

**Comment by kazup01** [comment]
Thanks @davesnx .

**Comment by rico345100** [comment]
@kazup01 Using MacOS 10.12.5.

**Comment by kazup01** [comment]
Thanks @rico345100 .

**Comment by pksh** [comment]
I had the same issue few days back. Boostnote did prompt me with the Initialise Storage screen but didn't thought through why I got this after an upgrade, which eventually wiped out my notes.  I really liked the easiness of using Boostnote but regretted not keeping a backup. Win7 user Boostnote 0.8.15 upgrade.

**Comment by larrypolk** [comment]
Having the same issue. Boostnote 0.8.15 on macOS 10.12.6. Last upgrade seems to have overwritten the contents of the notes directory. Fortunately, I backed up my notes before upgrading.

**Comment by johnbreau** [comment]
Every time I update the application I lose the link to all my notes and then have to spend the time to relink. This has happened several times. Yesterday, I was forced to update again. Now the relinking doesn't even work. I liked the UI and functionality of this application, but I'm tired of fighting these updates.

**Comment by mugshepherd** [comment]
experiencing the same problem.  

Windows 10 Home.  
Boostnote 0.8.15

I can see where the cson files are, but I'm not able to get Boostnote to recognize them.  They're contained in [user]/Boostnote/notes. I select that folder and the application creates a /notes folder within it (so it's linked to /Boostnote/notes/notes.  I then select /Boostnote/, and it says it's linked to the /Boostnotes/notes folder, but the notes included are empty - it doesn't grab any of them.   I placed some important code in a note just the day before the upgrade and now can't seem to access it.  Shame - it's a nice little application.

**Comment by rico345100** [comment]
@mugshepherd That's exactly same with me. I have the cson files but tried to load them results nothing.

**Comment by larrypolk** [comment]
@mugshepherd @rico345100 Same thing for me. I didn't spell it out in my earlier comment but that is the exact same behavior I'm seeing.

**Comment by david-hollifield** [comment]
+1 
Windows 10 Pro
Boostnote 0.8.15

Files still there on drive, just can't get them to show up. Tried re-linking, removing/adding storage, no luck after updating to 0.8.15

**Comment by vilmes21** [comment]
+ 1. Tried to add storage to my folder with all my previous .cson files. Still nothing shows up. This kind of bug will cause desastrious loss of your users.

**Comment by bjmarr** [comment]
Firstly apologies on the length of this post, it kind of got out of hand as I was testing a few things.

I've just encountered this problem and maybe have a solution for those struggling with this.
Things that I know prior to the issue:
- I'm on Windows 10 Pro (creators update) and version 0.8.18 of Boostnote.
- I have recently added some large markdown notes.
- I dragged a note from one folder to another (within the Boostnote UI).
- I created a second storage folder for another project.

When I opened Boostnote I was presented with only 2 notes out of 50+ and the trash was empty.

I navigated to the CSON files as suggested above and verified all of the data was present in the files. I checked the boostnote.json and the hashes within that file matched the individual CSON note files.
I then poked around (using Notepad++) the most recent notes to see if there was anything different and I noticed that some of the newest ones were missing data from the top of the file:

Newest files:
`type: "MARKDOWN_NOTE"
content: '''`

All of the older files:
`type: "MARKDOWN_NOTE"
folder: "e426e73a60c6f2be2752"
title: "Notes"
content: '''`

The missing information was put into the footer alongside the creation dates/star status but only for those newer files:
`'''
title: "Notes 2"
folder: "f1dc007582502d2f7111"
createdAt: "2017-12-20T17:55:53.329Z"
updatedAt: "2017-12-20T17:55:53.329Z"`

I simply moved these files out of the folder and then imported the data storage again and it worked. I have no idea what exactly triggered it as I made several changes all in one go (I was reorganising my notes).

I made alterations to one of the newer files and placed it back into the folder and it bugged Boostnote out again. I then discovered that there was a `'''` in the middle of the document, I removed these and the note still wouldn't show up. However there was a difference this time, all of my older notes were visible but the new one was not.
I then copied the headers/footers from a known working file and edited the titles/folder assignment hash and it now works absolutely fine. The long note is playing ball and Boostnote is working as expected.

**Comment by fi0** [comment]
Same issue on 0.8.19 
macOS 10.12.6

**Comment by skittlyoaten** [comment]
For whatever reason, after the update I lost all mine as well.  On install it made a new folder for me in my home folder.  and I couldn't seem to find my old one.  What I found, was on the old install, my Boostnote folder was not called Boostnote, it was actually my home folder directly.. i.e.  I had:   ~/boostnote.json and a ~/notes/ folder, which contained all my old notes directly in ~/.

I just copied the .json file into the newly created ~/Boostnote folder and overwrote the one in there.  Then moved all the notes into the newly created notes folder.  everything seems to be working now.  except of course, my preferences are all gone.

**Comment by chriskr7** [comment]
After I updated 0.8.20, my all notes are missing. So I tried to link storage where my cson files are.
After linking, It only shows the note name not files not folders. 
So I tried to create new folder and link it. Weird thing is that after I quit boostnote and 
start it again, all linked note names are gone.
Also, I was using boostnote mobile app through dropbox. I can see notes from my mobile app.
Plz tell me how to fix this :(

**Comment by jowj** [comment]
This seems to be same behavior I'm seeing. When I install new on Win10 pro with version 0.8.20 of boostnote i get no default starting storage.

I can add storage locally which works fine. 
I cannot add storage to the c:\users\me\dropbox\apps\boostnote-mobile\ directory though; i click create and it never says its new storage.

then, when i quit boostnote and relaunch none of my local storage shows up either. the files are still there, of course, but the "storage" link to the boostnote app is broke.

Let me know if I can give any more information / if there's a way to fix this.

**Comment by dotpyfe** [comment]
bumping this issue - update to v0.10.0 deleted all of my notes, i searched my drive for any cson files and was unable to find anything

**Comment by Rokt33r** [comment]
@jowj I assume your data is corrupted by some reason. Could you take a screenshot of console? You can open it by pressing `Cmd + Alt + I`, just same as Chrome.

**Comment by Rokt33r** [comment]
@dotpyfe Which version were you using before updating?

**Comment by dotpyfe** [comment]
@Rokt33r I'm not sure, presumable the directly preceding version. I had not manually initiated the update, it was applied automatically

**Comment by jowj** [comment]
@Rokt33r the issue fixed itself ~5 days later and I forgot to report back. ~5 days later I was tabbing between boostnote and chrome and: my Dropbox storage AND all the storage I tried to add but couldn't showed up at all once. My best guess was some sort of cache issue that i didn't catch and persisted through reboot.

**Comment by Rokt33r** [comment]
I think I have to close this issue although it is very critical. Lots of cases, I've seen, are quite different between users.

@baegmon Could you create another issue and tell us which version you'd used?

**Comment by salexln** [comment]
Any update on this issue??
This just happened to my and all my notes are gone!

Version 0.15.3 (0.15.3)
macOs Catalina 10.15.5 (19F101)

**Comment by Flexo013** [comment]
@salexln I recommend checking out this thread as it has some suggestions: https://github.com/BoostIO/Boostnote/issues/3575

If you still need help please comment there, as is it a more up to date thread.

**Comment by incace** [comment]
So, this just happened to me on v0.16.1, macOS 10.13.6. Opened Boostnote, the entire app is reset to default with none of my previous storage locations. I have a backup of my notes so that is no big deal, but the irritating part for me is all my preferences have been wiped. This included a custom CSS theme for preview I've been working off and on for about 6 months. I have some backed up versions of it, but not the most recent. Could there be any sort of backup of the preferences floating around on my system somewhere?

**Comment by rafaeltovar** [comment]
The same for me. All notes are gone. I lost few years of notes. It is difficult to decide if I will continue using this app. :(

Version 0.21.1
macOS 10.14.6 (Mojave)

Any `.cson` file found.

**Comment by Komediruzecki** [comment]
Hi, rafaeltovar,

if you updated to a new boost note next, those two applications are not compatible (boost note legacy and next).

Can you please let me know if you were using boost note legacy or boost note next prior to the update?

The version you specified is for boost note next, boost note legacy does not have such high versions.

If you updated to boost note next, did you update to local space application or cloud?

In any case, you cannot use or import storage containing *.CSON notes, only *.JSON file system storage is available in boost note next storages. 

For boost note cloud, you cannot have file system storage whatsoever. 

Can you describe your whole procedure of updating, with screenshots possibly, so we can see where things went wrong?

In short, if you had legacy storage with *.CSON notes, the notes should be on your disk, the boost note next will not erase them, and if you tried to import that storage into boost note next, the storage will complain of an invalid boostnote.json structure.

Can you provide me the details of your current `boostnote.json` so I can see what are you using, and more specifically what happened?

Also any errors from the application developer tools would be great (`ctrl+shift+i` to open)

**Comment by rafaeltovar** [comment]
Hi @Komediruzecki, thanks for the response.

yesterday I reboot my macbook and the app was upgraded (with the new icon). When I opened the app this asked me to create an account. I made the account and when I went to the app all my notes are gone!

Before I uninstalled Boost Note and  I installed the local version of the app, but my notes was gone already. :(

I saw that exist a folder in `~/Library/Application Support/Boost Note` and `~/Library/Application Support/Boost Note - Local`. I deleted the `Local` one and create a link to `~/Library/Application Support/Boost Note`. I restarted the local app but nothing, my notes are gone forever... :'(

**Comment by Komediruzecki** [comment]
Hi,

I am not aware of any auto update procedure, but not sure if for MacOS there is auto update.

For the installations, can you link the releases you installed (cloud and local space), also can you confirm if you had legacy boost note or just boostnote next, I suppose boostnote next since legacy didn't receive updates. 

If you installed boostnote local spaces, the latest application comes with no spaces and you need to manually load the space. Please backup your data on those locations before doing anything like this.

Then you can try loading it into local spaces if you have boostnote next local space, can you describe the contents of folders you mention?

If inside you have folders notes and attachments and file boostnote.json, can you describe, screenshot the contents of this and boostnote.json?

If the boost note notes folder contains *. cson files you should not load the location in latest boost note local spaces (you will get error), if it contains your *. json notes, please backup it and then try loading in boost note local spaces. By naming a space and giving the folder parent location you mentioned above if it contains the ecessary file boostnote.json, otherwise the storage will be created at that location, so be careful to backup any data prior to this

Let me know if you have any questions or need further assistance on any of the procedures.
